compiler/backend/src/org/jetbrains/kotlin/codegen/intrinsics/IntrinsicMethods.java

```diff
@@ -143,6 +143,9 @@ public IntrinsicMethods(JvmTarget jvmTarget, boolean shouldThrowNpeOnExplicitEqu
         declareIntrinsicFunction(FQ_NAMES.string, "plus", 1, new Concat());
         declareIntrinsicFunction(FQ_NAMES.string, "get", 1, new StringGetChar());
 
+        intrinsicsMap.registerIntrinsic(TEXT_PACKAGE_FQ_NAME, FQ_NAMES.string, "trimMargin", 1, new TrimMargin());
```

Comment:
We've discussed internally the idea to apply the intrinsic unconditionally and figured that it might lead to compatibility problems in case we decide to change the behavior of `trimMargin`/`trimIndent` on some corner case in a future release. It seems that these kinds of problems will only arise if the compiler of version X is compiling code against standard library of version Y where Y > X: the user expects behavior of version Y at runtime, but will actually observe behavior of version X because that's what was executed in the compiler at compilation time.

Therefore we'd like to restrict these intrinsics to be applicable only if the version of the standard library in dependencies is _not greater_ than the version of the compiler itself. This can be achieved by checking if the API version (can be obtained via `languageVersionSettings.apiVersion` at the call site) is <= than the version of the standard library in the runtime of the compiler (can be obtained via `KotlinVersion.CURRENT`). (Sorry if this sounds confusing, I can do it myself if it's unclear, just wanted to notify you that we'd like it to work that way -- maybe you have some other related concerns.)

Reply:
Makes sense. Added this check. Had to use `ApiVersion.parse(KotlinVersion.CURRENT.toString())` since those were of different types. Let me know if there's some other way to compare these two.